packages/zoe/src/instanceRecordStorage.js

```diff
@@ -0,0 +1,102 @@
+// @ts-check
+
+import { assert, details as X, q } from '@agoric/assert';
+import { assertKeywordName, getKeywords } from './cleanProposal';
+
+/**
+ * @callback AddIssuerToInstanceRecord
+ *
+ * Add an issuer and its keyword to the instanceRecord for the
+ * contract instance
+ *
+ * @param {Keyword} keyword
+ * @param {IssuerRecord} issuerRecord
+ * @returns {void}
+ */
+
+/**
+ * @callback GetInstanceRecord
+ * @returns {InstanceRecord}
+ */
+
+/**
+ *
+ * @typedef {{ addIssuerToInstanceRecord: AddIssuerToInstanceRecord,
+ * getInstanceRecord: GetInstanceRecord, getTerms: () => Terms,
+ * getIssuers: () => IssuerKeywordRecord,
+ * getBrands: () => BrandKeywordRecord, assertUniqueKeyword: (keyword:
+ * Keyword) => void }} InstanceRecordManager
+ */
+
+/**
+ *
+ * @param {InstanceRecord} instanceRecord
+ * @returns {InstanceRecordManager}
+ */
+export const makeInstanceRecordStorage = instanceRecord => {
+  const addIssuerToInstanceRecord = (keyword, issuerRecord) => {
+    instanceRecord = {
+      ...instanceRecord,
+      terms: {
+        ...instanceRecord.terms,
+        issuers: {
+          ...instanceRecord.terms.issuers,
+          [keyword]: issuerRecord.issuer,
+        },
+        brands: {
+          ...instanceRecord.terms.brands,
+          [keyword]: issuerRecord.brand,
+        },
+      },
+    };
+  };
+
+  const getInstanceRecord = () => harden(instanceRecord);
+  const getTerms = () => instanceRecord.terms;
+  const getIssuers = () => instanceRecord.terms.issuers;
+  const getBrands = () => instanceRecord.terms.brands;
+
+  const assertUniqueKeyword = keyword => {
+    assertKeywordName(keyword);
+    assert(
+      !getKeywords(instanceRecord.terms.issuers).includes(keyword),
+      X`keyword ${q(keyword)} must be unique`,
+    );
+  };
+
+  return {
```

Comment:
Missing harden, unless you're leaving it mutable for a reason. If so, please add a comment here explaining that reason.

```suggestion
  return harden({
```

Reply:
Fixed